pkg/task/task.go

```diff
@@ -0,0 +1,197 @@
+/*
```

Comment:
Seems perfect to me. @jcooklin Thanks for this work Joel
Is there a way we can substitute your PR instead of mine for this to get integrated ?